spec/actions_specs/xcodebuild_spec.rb

```diff
@@ -260,7 +262,70 @@
           + "-scheme \"MyApp\" " \
           + "-workspace \"MyApp.xcworkspace\" " \
           + "build " \
-          + "| xcpretty --color --simple"
+          + "| tee '#{File.expand_path("~/Library/Logs/fastlane/xcbuild/#{Time.now.strftime('%F')}/#{Process.pid}/xcodebuild.log")}' | xcpretty --color --simple"
+        )
+      end
+    end
+
+    describe "xcbuild without xpretty" do
+      it "is equivalent to 'xcodebuild build'" do
+        result = Fastlane::FastFile.new.parse("lane :test do
+          xcbuild(
+            scheme: 'MyApp',
+            workspace: 'MyApp.xcworkspace',
+            raw_buildlog: true
+          )
+        end").runner.execute(:test)
+
+        expect(result).to eq(
+          "set -o pipefail && " \
+          + "xcodebuild " \
+          + "-scheme \"MyApp\" " \
+          + "-workspace \"MyApp.xcworkspace\" " \
+          + "build " \
+          + "| tee '#{File.expand_path("~/Library/Logs/fastlane/xcbuild/#{Time.now.strftime('%F')}/#{Process.pid}/xcodebuild.log")}' "
+        )
+      end
+    end
+
+    describe "xcbuild without xpretty and with test" do
+      it "is equivalent to 'xcodebuild build'" do
+        result = Fastlane::FastFile.new.parse("lane :test do
+          xcbuild(
+            scheme: 'MyApp',
+            workspace: 'MyApp.xcworkspace',
+            raw_buildlog: true,
+            test: true
+          )
+        end").runner.execute(:test)
+
+        expect(result).to eq(
+          "set -o pipefail && " \
+          + "xcodebuild " \
+          + "-scheme \"MyApp\" " \
+          + "-workspace \"MyApp.xcworkspace\" " \
+          + "build test " \
+          + "| tee '#{File.expand_path("~/Library/Logs/fastlane/xcbuild/#{Time.now.strftime('%F')}/#{Process.pid}/xcodebuild.log")}' "
+        )
+      end
+    end
+
+    describe "xcbuild without xpretty and with test and reports" do
+      it "is equivalent to 'xcodebuild build'" do
+        result = Fastlane::FastFile.new.parse("lane :test do
+          xcbuild(
+            scheme: 'MyApp',
+            workspace: 'MyApp.xcworkspace',
+            raw_buildlog: true,
+            report_formats: ['html'],
+            test: true
+          )
+        end").runner.execute(:test)
+
+        expect(result).to eq(
+          "set -o pipefail && " \
+          + "cat '#{File.expand_path("~/Library/Logs/fastlane/xcbuild/#{Time.now.strftime('%F')}/#{Process.pid}/xcodebuild.log")}' " \
```

Comment:
Operator `+` should be surrounded with a single space.<br>Do not use `Time.now.strftime` without zone. Use one of `Time.zone.now.strftime`, `Time.now.strftime.current`, `Time.now.strftime.in_time_zone`, `Time.now.strftime.utc`, `Time.now.strftime.getlocal`, `Time.now.strftime.iso8601`, `Time.now.strftime.jisx0301`, `Time.now.strftime.rfc3339`, `Time.now.strftime.to_i`, `Time.now.strftime.to_f` instead.